[Fix] Elemental mobs physical resistances and detection in Central Temenos 2F and Eastern tower

kelppo opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
Before this commit, the Elementals in Central Temenos - 2nd Floor take 0 physical damage because of how they were set up, fix this by flipping the sign of xi.mod.UDMGPHYS and xi.mod.UDMGMAGIC to match scripts/battlefields/Temenos/temenos_eastern_tower.lua and update the comment in there.
<!-- Describe what your PR does here. If it closes an existing issue, you can mention: "Closes #1234" and GitHub will link this PR to that issue. -->

## Steps to test these changes
1. !pos 580.000 -2.375 104.000 37
2. !addkeyitem white_card
3. !addkeyitem cosmo_cleanse
4. Enter Central Temenos - 2nd Floor, observe how the elementals take no physical damage
5. Apply this patch
6. Observe how they now take physical damage

<!-- Clear and detailed steps to test your changes here -->

Xaver-DaRed commented:
Ok,so...
```c++
        float resist = 1.f + PDefender->getMod(Mod::UDMGPHYS) / 10000.f;
        resist       = std::max(resist, 0.f);
        damage       = (int32)(damage * resist);

        resist = 1.f + PDefender->getMod(Mod::DMGPHYS) / 10000.f + PDefender->getMod(Mod::DMG) / 10000.f;
        resist = std::max(resist, 0.5f);                        // PDT caps at -50%
```

Translation: regular DMGPHYS + DMG are capped at 50% and multiplicative to UPHYSDMG.
This is a mess already, because we also have this...
```c++
    SLASH_SDT  = 49, // Slash Damage Taken
    PIERCE_SDT = 50, // Piercing Damage Taken
    IMPACT_SDT = 51, // Impact Damage Taken
    HTH_SDT    = 52, // Hand-To-Hand Damage Taken
```

This are the values in database for its physical resistance by dmg type
```sql
INSERT INTO `mob_resistances` VALUES (99,'Elemental-Air',0.25,0.25,0.25,0.25,0,0,0,0,0,0,0,0,0,0,-3,11,11,0,0,0,0);
```

Abilities that affect this modifiers are supposed to work on this mobs though, so I propose that, since the elementals at both Temenos are supposed to take 1/2 damage from physical, instead of  their regular 1/4fucking arround with UDMGPHYS, lets just set the dmgtype modifiers to a 1/2 to override the originals.

So
```lua
[xi.mod.SLASH_SDT   ] = 500,
[xi.mod.PIERCE_SDT  ] = 500,
[xi.mod.IMPACT_SDT ] = 500,
[xi.mod.IMPACT_SDT ] = 500,
```
Yes, those are the actual values, those modifier use a different scale.

This, regarding physical damage

Xaver-DaRed reviewed (changes requested):
Read the comments I left and apply them to the PR.
Or propose a better method. That works aswell.

Also, the comments in the code should be updated to reflect the new way of handling (removing mention to "doubling damage") since this modifiers are now being set to their proper values instead of applying a multiplicative modifier on top.

jmcmorris commented:
@Xaver-DaRed Much better solution than I had. I didn't consider changing the SDT directly while doing this. Nice work! 👍

kelppo commented:
PR updated, I re-tested both East and 2nd Floor and they take approximately the same damage with Xaver's changes.

TeoTwawki commented on scripts/battlefields/Temenos/temenos_eastern_tower.lua:
```diff
@@ -279,12 +279,15 @@ content.groups =
+            [xi.mod.PIERCE_SDT  ] = 500,
+            [xi.mod.IMPACT_SDT  ] = 500,
+            [xi.mod.HTH_SDT     ] = 500,
+            [xi.mobMod.DETECTION] = xi.detects.HEARING,
```
Detection change ~~and fast cast change~~ weren't mentioned in the commit or the pr description. If accurate to retail, they should have been mentioned.

_partially fooled by the highlighting_

kelppo replied:
I can't test on retail but ffxiclopedia says that they only detect sound in East.
https://ffxiclopedia.fandom.com/wiki/Fire_Elemental
Should we leave it like before this PR? Only the Elementals in East detect sound.

kelppo replied:
Never mind, before this PR only 2nd Floor Elementals detect sound, just let me know how to go about this.

TeoTwawki replied:
Next time just add it to your commit message. We can add it in if we squash your commits, but I don't like editing people messages.

TeoTwawki replied:
> Next time just add it to your commit message. We can add it in if we squash your commits, but I don't like editing people messages.

I thought this was pretty clear. I was **not** looking for another commit removing it after a commit adding it. I just want commit messages to reflect everything happening in a commit.

kelppo committed:
Fix Elemental resistance in Central Temenos - 2nd Floor

Before this commit, the Elementals in Central Temenos - 2nd Floor take 0
physical damage because they were given -100% xi.mod.UDMGPHYS, as suggested,
fix this by replacing xi.mod.UDMGPHYS with damage specific SDT, remove
xi.mod.UDMGMAGIC, add sound detection to East floor (TODO: verify this on
retail), and update comments.

Xaver-DaRed commented:
This one is good to go
